pytorch_lightning/trainer/data_loading.py

```diff
@@ -223,6 +246,23 @@ def replace_sampler(self, dataloader: DataLoader, sampler, mode: Optional[Runnin
                     f"`{dataloader_cls_name}(dataset, sampler=DistributedSampler(dataset))`."
                 )
 
+        # wrap the ``IterableDataset`` into a ``CaptureIterableDataset`` to record sampler states.
+        if _fault_tolerant_enabled() and isinstance(dl_kwargs["dataset"], IterableDataset):
+            # force the seed in the ``Dataset``
+            seed = int(os.getenv("PL_GLOBAL_SEED", 0)) + self.current_epoch + self.global_rank
```

Comment:
Do we want to keep this ?

Reply:
potentially saver to remove until we have a good POC with random state management for restarting.

Reply:
from an earlier discussion: https://github.com/PyTorchLightning/pytorch-lightning/pull/8366#discussion_r668056865
thoughts @justusschock @carmocca ?